### PR TITLE
update pyrodigal arguments to v 3.3.0

### LINKED ISSUE #5128: Pyrodigal: update the module output formats
### Have you checked the docs?

- [X] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [X] [nf-core modules documentation](https://nf-co.re/docs/contributing/modules)

### Description of the bug

Pyrodigal at the moment is pinned to outputting only `.gff`, however in the new version 3.3.0 (the current version) output format can be changed by adding the parametre `-f ${output_format}` . 

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by Darcy220606**
## PR checklist

This PR updates pyrodigal arguments to allow it to output different formats not only `gff`. This is now applicable in version 3.3.0

Closes #5128 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Review by louperelo — APPROVED** [review]
Looks good to me, but a check is not passing - Conda ....
I just think `gene_annotations` is a bit long for an emit tag, but I don't really have a good alternative, maybe `annot` or `annotations`?

**Comment by Darcy220606** [comment]
> Looks good to me, but a check is not passing - Conda .... I just think `gene_annotations` is a bit long for an emit tag, but I don't really have a good alternative, maybe `annot` or `annotations`?

Thanks @louperelo ! I changed it to annotations instead. I dont know whats the issue with conda failing with CI tests on github, because locally its running on all three docker, conda and singularity without failing. Any suggestions what to do here?

File: modules/nf-core/pyrodigal/meta.yml [hunk 1]
```diff
@@ -23,6 +23,9 @@ input:
       type: file
       description: FASTA file
       pattern: "*.{fasta.gz,fa.gz,fna.gz}"
+  - output_format:
+      type: string
+      description: Output format ("gbk"/"gff")
```
Comment (jasmezz) [inline modules/nf-core/pyrodigal/meta.yml:28]:
```suggestion
      description: Output format
      pattern: "{gbk,gff}"
```

File: modules/nf-core/pyrodigal/meta.yml [hunk 2]
```diff
@@ -33,10 +36,10 @@ output:
       type: file
       description: File containing software versions
       pattern: "versions.yml"
-  - gff:
+  - annotations:
       type: file
-      description: gene annotations in gff format
-      pattern: "*.{gff.gz}"
+      description: gene annotations in output format given as input
```
Comment (jasmezz) [inline modules/nf-core/pyrodigal/meta.yml:41]:
```suggestion
      description: gene annotations. The file format is specified via input channel "output_format".
```
Just updated the wording, I was confused by "gene annotations in output format given as input"

File: modules/nf-core/pyrodigal/meta.yml [hunk 3]
```diff
@@ -33,10 +36,10 @@ output:
...
+      pattern: "*.{output_format}.gz"
```
Comment (jasmezz) [inline modules/nf-core/pyrodigal/meta.yml:42]:
```suggestion
      pattern: "*.{gbk,gff}.gz"
```

**Review by jasmezz — CHANGES_REQUESTED** [review]
Pigz is apparently missing in the CI test environment, so I added is via the previous commit. Here some more suggestions to clarify the meta.yml. But otherwise ready to go :rocket:

**Review by jasmezz — APPROVED** [review]

**Pull request merged**